app/views/questionnaire.py

```diff
@@ -189,8 +191,21 @@ def get_block_handler(
     list_store = questionnaire_store.list_store
     metadata = questionnaire_store.metadata
     answer_store = questionnaire_store.answer_store
-
     current_location = Location(block_id, list_name, list_item_id)
+    questionnaire_store_updater = QuestionnaireStoreUpdater(
+        current_location, schema, questionnaire_store, block_id
+    )
+
+    if current_location in questionnaire_store.completed_items.block_locations:
+        questionnaire_store_updater.remove_completed_block(current_location)
+
+    try:
+        current_section_id = schema.get_section_via_block_id(block_id)['id']
+    except TypeError:
+        current_section_id = schema.json.get('sections')[0]['id']
+
+    if current_section_id in questionnaire_store.completed_items.sections:
+        questionnaire_store_updater.remove_completed_section(current_section_id)
```

Comment:
State should not be changed on a GET request. Why does a GET need to remove completeness?